shell.nix

```diff
@@ -0,0 +1,17 @@
+{ system ? builtins.currentSystem }:
+let
+  # nixpkgs / devshell is only used for development. Don't add it to the flake.lock.
+  devshellGitRev = "709fe4d04a9101c9d224ad83f73416dce71baf21";
+
+  nixpkgsSrc = ./.;
+
+  devshellSrc = fetchTarball {
+    url = "https://github.com/numtide/devshell/archive/${devshellGitRev}.tar.gz";
+    sha256 = "1px9cqfshfqs1b7ypyxch3s3ymr4xgycy1krrcg7b97rmmszvsqr";
+  };
+
+  pkgs = import nixpkgsSrc { inherit system; };
+  devshell = import devshellSrc { inherit system pkgs; };
```

Comment:
I see a fundamental problem here: There is no way to introduce breaking changes in `nixpkgs` that touch `devshell` without patching the latter along. Otherwise the shell will simply break. We just added quite an arbitrary dependency.

Technically correct solution would be to pass the historic version of `nixpkgs` that `devshell` was developed against, but that's quite expensive.

Reply:
> Technically correct solution would be to pass the historic version of nixpkgs that devshell was developed against, but that's quite expensive.

I can see the problem, but for the reasonable future, I think `devshell` will maintain 

> We should consider making it optional though, such as by only providing a menu command, or keeping it really really short, with a reference to more detailed instructions. Just imagine regular developers seeing it every time they cd nixpkgs.

It happens so that invoking via `nix-shell` doesn't expose the menu. We can also lobby upstream for a documented solution. If this get's merged, such solution should emerge within a matter of weeks.

Reply:
> Technically correct solution would be to pass the historic version of nixpkgs that devshell was developed against, but that's quite expensive.

I can see the problem. For the reasonable future, I think `devshell` will be well maintained and any breakage will be resolved within days, not weeks.

The problem manifests once/if (core) maintainers or the CI becomes heavily dependent on it. By then, devshell, whose stated objective is to completment `mkShell` here in nixpkgs, will either be already upstreamed or heavily maintained.

Historical nixpkgs, without further guardrails, would imply historical nixpkgs-fmt, historical nixpkgs-review, etc, etc.

I think cost benefit tips slightly towards accepting the risk that it (might evnetually maybe) break. ... and fix it when we get there.